persistence/src/main/scala/hmda/persistence/FilingPersistence.scala

```diff
@@ -0,0 +1,89 @@
+package hmda.persistence
+
+import akka.actor.{ ActorLogging, ActorRef, ActorSystem, Props }
+import akka.persistence.{ PersistentActor, SnapshotOffer }
+import hmda.model.fi.Filing
+import hmda.persistence.CommonMessages._
+import hmda.persistence.FilingPersistence._
+
+object FilingPersistence {
+
+  case class CreateFiling(filing: Filing) extends Command
+  case class UpdateFilingStatus(filing: Filing) extends Command
+  case class GetFilingByPeriod(period: String) extends Command
+
+  case class FilingCreated(filing: Filing) extends Event
+  case class FilingStatusUpdated(filing: Filing) extends Event
+
+  def props(fid: String): Props = Props(new FilingPersistence(fid))
+
+  def createFilings(fid: String, system: ActorSystem): ActorRef = {
+    system.actorOf(FilingPersistence.props(fid))
+  }
+
+  case class FilingState(filings: Seq[Filing] = Nil) {
+    def updated(event: Event): FilingState = {
+      event match {
+        case FilingCreated(f) =>
+          FilingState(f +: filings)
+        case FilingStatusUpdated(modified) =>
+          val x = filings.find(x => x.id == modified.id).getOrElse(Filing())
+          val i = filings.indexOf(x)
+          FilingState(filings.updated(i, modified))
+      }
+    }
+  }
+}
+
+// Filings per Institution (fid = institution identifier)
+class FilingPersistence(fid: String) extends PersistentActor with ActorLogging {
+
+  var state = FilingState()
+
+  def updateState(e: Event): Unit = {
+    state = state.updated(e)
+  }
+
+  override def preStart(): Unit = {
+    log.debug(s"Filings started at ${self.path}")
+  }
+
+  override def persistenceId: String = s"filings-$fid"
+
+  override def receiveRecover: Receive = {
+    case e: Event => updateState(e)
+    case SnapshotOffer(_, snapshot: FilingState) =>
+      log.debug(s"Recovering from snapshot")
+      state = snapshot
+
+  }
+
+  override def receiveCommand: Receive = {
+    case CreateFiling(f) =>
+      if (!state.filings.contains(f)) {
+        persist(FilingCreated(f)) { e =>
+          log.debug(s"Persisted: $f")
+          updateState(e)
+        }
+      }
+
+    case UpdateFilingStatus(modified) =>
+      if (state.filings.map(x => x.id).contains(modified.id)) {
```

Comment:
This pattern is common to all of the persistence actors. Should this `if` (and others like it) have an `else`? 

If I understand correctly, this line is preventing the `UpdateFilingStatus` method from trying to update a nonexistent record. If this case does receive an incorrect `id`, it seems like this will fail silently. Would it be better to log that error, for debugging purposes?